source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/GenerateFormContainer.tsx

```diff
@@ -14,16 +22,155 @@ export interface IGenerateFormContainerProps {
+      loading: loadingAcquisitionFileOwners,
+    },
+  } = useAcquisitionProvider();
+  const {
+    getAcquisitionInterestHolders: {
+      execute: fetchInterestHolders,
+      loading: loadingInterestHolders,
+    },
+  } = useInterestHolderRepository();
 
-  const generateH0443 = useGenerateH0443();
+  const fetchAllRecipients = useCallback(async () => {
+    let i = 0;
+    const generateRecipientsList: LetterRecipientModel[] = [];
 
-  const generateH120 = useGenerateH120();
+    const file = await getAcquisitionFile(acquisitionFileId);
+    if (file === undefined) {
+      return;
+    }
+
+    const fileOwnersFetchCall = retrieveAcquisitionFileOwners(acquisitionFileId);
+    const interestHoldersFetchCall = fetchInterestHolders(acquisitionFileId);
+
+    // Add owners and interest holders
+    await Promise.all([fileOwnersFetchCall, interestHoldersFetchCall]).then(
+      ([fileOwners, intHolders]) => {
+        if (fileOwners !== undefined) {
+          const ownersLetterModel = fileOwners?.map(owner => new Api_GenerateOwner(owner)) ?? [];
+          const ownerRecipientsLetterModel =
+            ownersLetterModel.map(x => new LetterRecipientModel(i++, 'OWNR', x)) ?? [];
+          generateRecipientsList.push(...ownerRecipientsLetterModel);
+        }
+
+        if (intHolders !== undefined) {
+          intHolders.map(async holder => {
+            if (holder.personId) {
+              const person = !!holder?.personId
+                ? (await getPersonConcept(holder?.personId))?.data
+                : null;
+
+              if (person) {
+                generateRecipientsList.push(
+                  new LetterRecipientModel(
+                    i++,
+                    'HLDR',
+                    Api_GenerateOwner.fromApiPerson(person),
+                    holder,
+                  ),
+                );
+              }
+            } else if (holder.organizationId) {
+              const org = !!holder?.organizationId
+                ? (await getOrganization(holder?.organizationId))?.data
+                : null;
+
+              if (org) {
+                generateRecipientsList.push(
+                  new LetterRecipientModel(
+                    i++,
+                    'HLDR',
+                    Api_GenerateOwner.fromApiOrganization(org),
+                    holder,
+                  ),
+                );
+              }
+            }
+          });
+        }
+      },
+    );
+
+    // Add owners solicitors to recipients list
+    if (file.acquisitionFileOwnerSolicitors && file.acquisitionFileOwnerSolicitors.length > 0) {
+      await Promise.all(
+        file.acquisitionFileOwnerSolicitors.map(async solicitor => {
+          if (solicitor.personId) {
+            const person = !!solicitor?.personId
+              ? (await getPersonConcept(solicitor?.personId))?.data
+              : null;
+
+            if (person) {
+              const personSolicitorModel = Api_GenerateOwner.fromApiPerson(person);
```

Comment:
seems like maybe this and the org version of this function could be extracted into a common function?

Reply:
I can't see any other scenario where this is used outside of the GenerateLetter context, and I think it's rather handy to have it next to the model.

Reply:
Owner solicitor is also used in the Generate Expropriation Forms. I talked to Dev and we agreed to keep using GeneratePerson for owner solicitor in those forms but there's a JIRA to use organizations as solicitors as well so this code can be reused

I think we would need to create a "GenerateContact" that can be either a GeneratePerson or a GenerateOrganization

No need to change this PR but just an FYI that we will need to make this "person or org" code more shareable

Reply:
Made changes to use the new InterestHolder schema.